alexandria/core/models.py

```diff
@@ -137,6 +137,22 @@ class Document(UUIDModel):
     marks = models.ManyToManyField(Mark, blank=True, related_name="documents")
     date = models.DateField(blank=True, null=True)
 
+    def clone(self):
+        latest_original = (
+            self.files.filter(variant="original").order_by("-created_at").first()
+        )
+        thumbnail = latest_original.renderings.first()
+
+        self.pk = None
+        self.save()
+
+        for file in [latest_original, thumbnail]:
+            file.pk = None
+            file.document = self
+            file.save()
+
+        return self
```

Comment:
Let's check if we actually get two files in minio when cloning. If not, we might run into problems when one of them is deleted.

Reply:
added code that copies the file in S3.